### PR TITLE
Add support for olmo

### PR DESCRIPTION
**Pull request by dakinggg**
Test run: `olmo-3-aCGqyY`

### CODE REVIEW

File: llmfoundry/models/hf/hf_fsdp.py [hunk 1]
```diff
@@ -141,15 +142,20 @@ def prepare_hf_causal_lm_model_for_fsdp(model: Union[PreTrainedModel,
...
     lm_head = model.get_output_embeddings()
-    # some models (OPT) implement .get_input_embeddings for the causal subclass
-    # but all of them implement it for the base model
-    tied_embeddings = causal_base_model.get_input_embeddings()
+    # Try to get input embeddings from the transformer backbone
+    # and then from the XXXForCausalLM
+    try:
+        tied_embeddings = causal_base_model.get_input_embeddings()
+    except:
+        tied_embeddings = model.get_input_embeddings()
```
Comment (mvpatel2000) [inline llmfoundry/models/hf/hf_fsdp.py:157]: why is this necessary?

Comment (dakinggg) [reply llmfoundry/models/hf/hf_fsdp.py:157]: cause models have this defined in different places

**Review by mvpatel2000 — APPROVED** [review]
